docs/migration/4_0.md

```diff
@@ -253,14 +253,40 @@ Lib: @spartacus/core
 Class: UserAddressService
 Change: Two new required constructor parameters `userAddressConnector: UserAddressConnector` and `command: CommandService`
 
+
+## New Checkout Library
+
+Spartacus 4.0 introduces the checkout library.  The checkout related code is moved out of `@spartacus/core` and `@spartacus/storefrontlib` into one of the checkout lib's entry points.  The checkout library is split into these entry points:
+
+```
+@spartacus/checkout/assets 
+( checout related i18n keys are moved here )
+
+@spartacus/checkout/components
+( checout related UI codee is moved here. This includes components, guards anD ui services )
+
+@spartacus/checkout/core
+The checkout facade API implementation are moved here, as well as connectors, event builder, event listener, models, other services, and state management.
+
+@spartacus/checkout/occ
+The checkout related OCC code is moved here. This includes the checkout related adapters and converters.
+
+@spartacus/checkout/root
+The root entry point is, by convention, meant to always be eager loaded.  It contains the config, events, facades, http interceptors and models.
```

Comment:
facade interfaces, right? the implementation still in `checkout core`

Reply:
This is somewhat hidden for those who use the libs.  Their dependency is on the facade from the root entry point.